llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -56111,11 +56133,50 @@ static SDValue combineConcatVectorOps(const SDLoc &DL, MVT VT,
       break;
     case X86ISD::PCMPEQ:
     case X86ISD::PCMPGT:
-      if (!IsSplat && VT.is256BitVector() && Subtarget.hasInt256() &&
+      if (!IsSplat && VT.is256BitVector() &&
```

Comment:
Is the `!IsSplat` desirable?

Reply:
Yes w2e want to avoid splats - a single VPCMPEQ/GT op + VINSERTF128 is always going to be preferable to a VINSERTF128+VCVTDQ2PS+VCMPPS